src/main/groovy/net/wasdev/wlp/gradle/plugins/tasks/AbstractTask.groovy

```diff
@@ -70,4 +101,201 @@ abstract class AbstractTask extends DefaultTask {
         return (project.liberty.userDir == null) ? new File(installDir, 'usr') : new File(project.liberty.userDir)
     }
 
+    //Sets directories for the various configuration files
+    protected void initializeConfigVariables(){
+        if(project.liberty.configDirectory != null && project.liberty.configDirectory.length() != 0){
+            configDirectory = new File(project.liberty.configDirectory)
+        }
+
+        if(project.liberty.configFile != null && project.liberty.configFile.length() != 0){
+            configFile = new File(project.liberty.configFile)
+        }
+        else{
+            configFile = new File("/src/main/liberty/config/server.xml")
+        }
+
+        if(project.liberty.bootstrapPropertiesFile != null && project.liberty.bootstrapPropertiesFile.length() != 0){
+            bootstrapPropertiesFile = new File(project.liberty.bootstrapPropertiesFile)
+
+        }
+        else{
+            bootstrapPropertiesFile = new File("/src/main/liberty/config/bootstrap.properties")
+        }
+
+        if(project.liberty.jvmOptionsFile != null && project.liberty.jvmOptionsFile.length() != 0){
+            jvmOptionsFile = new File(project.liberty.jvmOptionsFile)
+        }
+        else{
+            jvmOptionsFile = new File("/src/main/liberty/config/jvm.options")
+        }
+
+        if(project.liberty.serverEnv != null && project.liberty.serverEnv.length() != 0){
+            serverEnv = new File(project.liberty.serverEnv)
+        }
+        else{
+            serverEnv = new File("/src/main/liberty/config/server.env")
+        }
+    }
+
+    //Gets the loose bootstrap and jvm properties from the build file
+    protected void initializeConfigProperties(){
+        if(project.liberty.bootstrapProperties != null && !project.liberty.bootstrapProperties.isEmpty()){
+            bootstrapProperties = project.liberty.bootstrapProperties
+        }
+        if(project.liberty.jvmOptions != null && !project.liberty.jvmOptions.isEmpty()){
+            jvmOptions = project.liberty.jvmOptions
+        }
+    }
+
+    private void copyFiles(File srcFile, File destFile){
```

Comment:
I would just use this java class for copying files - https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html
There are a bunch of great statics for performing operations on Files